### PR TITLE
chore: add renderers for TS - enum and interface

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

As in title:
- Add renderer for enum in TS
- Add renderer for interface in TS
- Add relevant unit tests

**Related issue(s)**
See also https://github.com/asyncapi/shape-up-process/issues/48

### CODE REVIEW

**Review by jonaslagoni — COMMENTED** [review]
Have a few clarification questions and general feedback. I think we should not focus on what is not available in the `CommonModel` class, example required properties and description. But JUST the bare minimum, even though it would make a better output. These kind of properties are something we have to add to the `CommonModel` at some point of course 😄

File: src/generators/javascript/TypeScriptRenderer.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      return 'number';
+    case 'boolean':
+      return 'boolean';
+    case 'array': {
+      const types = model.items ? this.renderType(model.items) : 'any';
+      return `Array<${types}>`;
+    }
+    default: return type || "any";
+    }
+  }
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply src/generators/javascript/TypeScriptRenderer.ts:48]: In first draft I had as first argument `string | string[]`, but then I realized than CommonModel treats items and properties as CommonModel instances, and by this I must also check `$ref`  property to render name of reference.  By splitting function to two separates, we make code more complex.

File: src/generators/javascript/TypeScriptRenderer.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  protected renderTypeSignature(type: CommonModel | CommonModel[], isOptional: boolean = false): string {
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply src/generators/javascript/TypeScriptRenderer.ts:50]: If type is written in `$ref` then we should render this ref, if type is array, we should render `Array<{items}>`. Putting as argument only `model.type` doesn't handle mentioned cases

File: src/generators/javascript/TypeScriptRenderer.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonaslagoni) [reply src/generators/javascript/TypeScriptRenderer.ts]: Fair enough 😄

File: src/generators/javascript/TypeScriptRenderer.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonaslagoni) [reply src/generators/javascript/TypeScriptRenderer.ts:50]: Fair 😄

**Review by jonaslagoni — APPROVED** [review]
👍

**Pull request merged**